src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/DataSetStep.test.tsx

```diff
@@ -0,0 +1,582 @@
+      within(featuredTables[1]).getByText(
+        'Test featured description 2 find me',
+      ),
+    ).toBeInTheDocument();
+
+    expect(
+      within(featuredTables[2]).getByRole('link', { name: 'Test featured 3' }),
+    ).toBeInTheDocument();
+    expect(
+      within(featuredTables[2]).getByText('Test featured description 3'),
+    ).toBeInTheDocument();
+  });
+
+  test('searching featured tables filters the list', async () => {
+    render(
+      <DataSetStep
+        {...wizardProps}
+        featuredTables={testFeaturedTables}
+        renderFeaturedTable={highlight => <a href="/">{highlight.name}</a>}
+        subjects={testSubjects}
+        onSubmit={noop}
+      />,
+    );
+
+    userEvent.click(
+      screen.getByRole('radio', { name: 'View all featured tables' }),
+    );
+
+    expect(
+      within(screen.getByTestId('featuredTables')).getAllByRole('listitem'),
+    ).toHaveLength(3);
+
+    userEvent.type(screen.getByLabelText('Search featured tables'), 'find me');
+    userEvent.click(screen.getByRole('button', { name: 'Search' }));
+
+    await waitFor(() => {
+      expect(screen.getByText('Clear search')).toBeInTheDocument();
+    });
+
+    expect(
+      within(screen.getByTestId('featuredTables')).getAllByRole('listitem'),
+    ).toHaveLength(1);
+  });
+
+  test('clicking `clear search` shows all featured tables', async () => {
+    render(
+      <DataSetStep
+        {...wizardProps}
+        featuredTables={testFeaturedTables}
+        renderFeaturedTable={highlight => <a href="/">{highlight.name}</a>}
+        subjects={testSubjects}
+        onSubmit={noop}
+      />,
+    );
+
+    userEvent.click(
+      screen.getByRole('radio', { name: 'View all featured tables' }),
+    );
+
+    expect(
+      within(screen.getByTestId('featuredTables')).getAllByRole('listitem'),
+    ).toHaveLength(3);
+
+    userEvent.type(screen.getByLabelText('Search featured tables'), 'find me');
+    userEvent.click(screen.getByRole('button', { name: 'Search' }));
+
+    await waitFor(() => {
+      expect(screen.getByText('Clear search')).toBeInTheDocument();
+    });
+
+    expect(
+      within(screen.getByTestId('featuredTables')).getAllByRole('listitem'),
+    ).toHaveLength(1);
+
+    userEvent.click(screen.getByRole('button', { name: 'Clear search' }));
+
+    await waitFor(() => {
+      expect(screen.queryByText('Clear search')).not.toBeInTheDocument();
+    });
+
+    expect(
+      within(screen.getByTestId('featuredTables')).getAllByRole('listitem'),
+    ).toHaveLength(3);
+  });
+
+  test('selecting a data set shows the details and featured tables', () => {
+    render(
+      <DataSetStep
+        {...wizardProps}
+        selectedRelease={testRelease}
+        featuredTables={testFeaturedTables}
+        renderFeaturedTable={highlight => <a href="/">{highlight.name}</a>}
+        subjects={testSubjects}
+        onSubmit={noop}
+      />,
+    );
+
+    userEvent.click(screen.getByLabelText('Subject 1'));
+
```

Comment:
Might be worth a simple check that the text from the previous view is no longer visible